src/Controller/Webhook/RefundOrderAction.php

```diff
@@ -49,10 +49,8 @@ public function __invoke(Request $request): Response
             return new JsonResponse(['error' => $exception->getMessage()], Response::HTTP_NOT_FOUND);
         }
 
-        /** @var StateMachineInterface $stateMachine */
-        $stateMachine = $this->stateMachineFactory->get($payment, PaymentTransitions::GRAPH);
-        if ($stateMachine->can(PaymentTransitions::TRANSITION_REFUND)) {
-            $stateMachine->apply(PaymentTransitions::TRANSITION_REFUND);
+        if ($this->getStateMachine()->can($payment, PaymentTransitions::GRAPH, PaymentTransitions::TRANSITION_REFUND)) {
+            $this->getStateMachine()->apply($payment, PaymentTransitions::GRAPH, PaymentTransitions::TRANSITION_REFUND);
```

Comment:
```suggestion
        $stateMachine = $this->getStateMachine();
        if ($stateMachine->can($payment, PaymentTransitions::GRAPH, PaymentTransitions::TRANSITION_REFUND)) {
            $stateMachine->apply($payment, PaymentTransitions::GRAPH, PaymentTransitions::TRANSITION_REFUND);
```
No point in resolving it twice